### ISSUE TITLE
Compiler warnings with 4.0.2 in SDL2

### ISSUE THREAD

**Issue by GitMensch** [Bill-Gray/PDCurses#57]
Environment:

~~~
$ uname -a
MINGW64_NT-10.0 MY-MACHINE 2.5.0(0.295/5/3) 2016-03-31 18:47 x86_64 Msys

$ cc --version
cc.exe (GCC) 4.8.1
~~~

Warnings:

~~~
 ../sdl2/pdcclip.c
../sdl2/pdcclip.c: In function 'PDC_getclipboard':
../sdl2/pdcclip.c:58:5: warning: implicit declaration of function 'strlen' [-Wimplicit-function-declaration]
     *length = strlen(*contents);
     ^
../sdl2/pdcclip.c:58:15: warning: incompatible implicit declaration of built-in function 'strlen' [enabled by default]
     *length = strlen(*contents);
               ^
cc -O2 -Wall  -I.. -Dmain=SDL_main -I./SDL2   -DHAVE_VSNPRINTF -DHAVE_VSSCANF -c ../sdl2/pdckbd.c
../sdl2/pdckbd.c: In function '_process_key_event':
../sdl2/pdckbd.c:258:17: warning: implicit declaration of function 'toupper' [-Wimplicit-function-declaration]
                 key = toupper(key);
                 ^
../sdl2/pdckbd.c: In function 'PDC_get_key':
../sdl2/pdckbd.c:401:9: warning: implicit declaration of function 'exit' [-Wimplicit-function-declaration]
         exit(1);
         ^
../sdl2/pdckbd.c:401:9: warning: incompatible implicit declaration of built-in function 'exit' [enabled by default]
../sdl2/pdcutil.c:23:24: warning: 'PDC_version' initialized and declared 'extern' [enabled by default]
 PDCEX PDC_version_info PDC_version = { PDC_PORT_SDL2,
                        ^
~~~

Additional note: this was after changing the Makefile to use

~~~
ifeq ($(NOPIC),Y)
	PIC = 
else
	PIC = -fPIC
endif

ifeq ($(DEBUG),Y)
	CFLAGS  = -g -Wall -DPDCDEBUG $(PIC)
else
	CFLAGS  += -O2 -Wall $(PIC)
endif
~~~

To prevent 

> warning: -fPIC ignored for target (all code is position independent) [enabled by default]

And running make with `make NOPIC=Y "EXE_EXTENSION=.exe"` (using UTF8=Y and/or WIDE=Y did not lead to any other error messages for the SDL port)

**Comment by Bill-Gray** [comment]
../sdl2/pdcclip.c:58:5: warning: implicit declaration of function 'strlen' [-Wimplicit-function-declaration]
     *length = strlen(*contents)

Try changing that to read...
     *length = (long)strlen(*contents)

../sdl2/pdckbd.c:258:17: warning: implicit declaration of function 'toupper' [-Wimplicit-function-declaration]
                 key = toupper(key);

   OK,  this one's a puzzlement.  `toupper` ought to require us to #include <ctype.h>.  'exit' ought to require #include <stdlib.h>.  Neither occurs in pdckbd.c.  Nevertheless,  I don't see that warning. I think that's because the SDL headers are bringing those in for me.

   It will not hurt to add,  in pdckbd.h,  after the #include <string.h> line,  a pair of lines to #include these two "missing" headers,  and it should cause the warnings to go away.

   Still looking into the issue with PDC_version.  Removing the PDCEX gets rid of the error,  and is the way it's done in Win32 console,  Win32a,  and X11.  I'm almost certain that's what needs to be done here,  too.

**Comment by GitMensch** [comment]
Fixed with 7aa0dd2.
